### DISCUSSION TITLE
Possible to get current count into a variable?

### DISCUSSION THREAD

**Discussion by jreviews** [spatie/ray#416] [Q&A] [answered]
I'd like to log some info at a specific count. When using `ray()->count()`is it possible to grab the current count value to use in a conditional? Using PHP btw.

**Comment by freekmurze** [comment, marked as answer]
You could do something like this `ray()->showIf(count($yourArray) > 3)`

**Reply by jreviews** [reply]
Thanks for the reply, but maybe I wasn't clear with my question. I'd like to do the `showIf` only when the internal ray counter  matches a number.  So something like this:

```php
ray()->count();
if (ray()->getCount() == 2 ) {
 // do or log something
}
```

**Reply by freekmurze** [reply]
Try this.

```
ray()->count('myCounter');
if (Spatie\Ray::getCounters()->get('myCounter')) ===2) {
}
```

If this works, I'd accept a PR that adds a function `counterValue` that returns the value of a counter.

**Reply by jreviews** [reply]
Thanks! For reference, this is what worked for me (not using Laravel). 

```php
\Spatie\Ray\Ray::$counters->get('myCounter')
```
If no one else submits the PR, I can try to do it myself if I can figure out the testing part.

```php
    public function counterValue(string $name)
    {
        return self::$counters->get($name);
    }
```

**Reply by patinthehat** [reply]
@jreviews I've submitted a [PR](https://github.com/spatie/ray/pull/437) for this feature.

**Reply by jreviews** [reply]
> @jreviews I've submitted a [PR](https://github.com/spatie/ray/pull/437) for this feature.

Thanks! that's great. Hopefully it gets merged.

**Reply by freekmurze** [reply]
Merged and release, thanks @patinthehat
